src/main.tsx

```diff
@@ -17,12 +18,14 @@ const router = createBrowserRouter([routesConfig]);
 
 ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
     <React.StrictMode>
-        <ContextProvider>
-            <InstallPromptProvider>
-                <IconContext.Provider value={iconConfig}>
-                    <RouterProvider router={router} />
-                </IconContext.Provider>
-            </InstallPromptProvider>
-        </ContextProvider>
+        <CategoryContextProvider>
+            <DayViewContextProvider>
```

Comment:
Day view context should only be available in the day view, it's irrelevant to other pages.